src/error-establisher.ts

```diff
@@ -0,0 +1,51 @@
+import { ErrorRequestHandler } from 'express';
+import i18next from 'i18next';
+import logger from './config/logger';
+import { isSystemError, AppError } from './util';
+
+export const sysErrorsHandler: ErrorRequestHandler = (
+  error,
+  req,
+  res,
+  next,
+) => {
+  logger.error(`[${error.name}]: ${error.logMsg ?? error.message}`);
+
+  if (isSystemError(error)) {
+    return void res.status(500).json({ message: i18next.t('server_error') });
+  }
+
+  if (error instanceof AppError) {
+    return void next(error);
+  }
+
+  if (error.status ?? error.statusCode) {
+    // any libraries using a `status` or `statusCode` mechanism as:
+    // i18next, passport.
+    /** @TestNote change the token manually */
```

Comment:
`@TestNote`? Is that a supported JSDoc field type?

Reply:
Oh boy, I forgot this, I usually have these comments as a `todo` or a self reminder note, This will be cleaned up.